### ISSUE TITLE
Extra closing parenthesis when having transparency in hsla()

### ISSUE THREAD

**Issue by georgechond94** [jhildenbiddle/css-vars-ponyfill#42]
Hi,

This is how my vars look like: 
```
--color-background-transparent-9: hsla(200, 20%, 25%, 0.9);
--color-background-tint: hsla(200, 15%, 90%, 1);
```

This is how I use them on an element:
```
background: var(--color-background-transparent-9, rgba(51, 68, 77, 0.9));
color: var(--color-background-tint, #e2e7e9);

```
This is how css-vars-ponyfill converts them:
```
background: hsla(200, 20%, 25%, 0.9)**)**;
color: hsla(197, 14%, 90%, 1);

```
Notice the extra closing parenthesis in the background property.

After a bit of investigation, it seems that the issue is the fallback value.
If I replace the **_rgba(51, 68, 77, 0.9)_** with **_#e2e7e9_** it works.

Does anyone have this issue before? Do you may know any workaround?

Thanks in advance!
George

**Comment by georgechond94** [comment]
Wrote a test and it fails on this one:
![image](https://user-images.githubusercontent.com/6044597/49148375-85358d00-f307-11e8-882c-85125a76635a.png)
![image](https://user-images.githubusercontent.com/6044597/49148468-c0d05700-f307-11e8-87b8-55e8ff0aad8c.png)

**Comment by lukasbisdorf** [comment]
I can confirm. Have the same issue.

**Comment by confraria** [comment]
i believe this is caused by balanced-match
line 315 in transform-css.js
```javascript
const varFuncData  = balanced('var(', ')', value);
```
experimented with a regexp but no luck 😞 

```
const balanced = require("balanced-match");
console.log(balanced('var(', /\)$/, 'var(--a, url(a))'))
```
https://runkit.com/confraria/5bfe9301611a28001298d44f
The ``varFuncData.post` is still `)`

**Comment by confraria** [comment]
yup it's balanced-match 
https://github.com/juliangruber/balanced-match/blob/master/index.js#L20
it's using  regexp to find a match, but then uses indexOf to find the position of the matched string, which then again finds the first `)`

**Comment by lukasbisdorf** [comment]
Yes - the problem is, that balanced did not know, that HSLa( is also a function, with its own closing ")". To make this Polyfills work correct, there is some kind of small interpreter needed how knows everything possible inside a var statement.

**Comment by jhildenbiddle** [comment]
Fixed in 1.15.3.

Thanks for the excellent description and debugging efforts. Made the fix much easier. 😄 

Let me know if the fix resolves your issues. I'll hold off closing the issue until then.

**Comment by confraria** [comment]
Thank you 🤗 All solved.
I had an issue with vars containing url(...).
works fine now!
